### PR TITLE
Fix SQL injection and potential bookSearch bug in book.js

### PR DESCRIPTION
**Pull request by togimoto**
Please check that the logic of the queries remain unchanged.

### CODE REVIEW

File: src/backend/controllers/book.js [hunk 1]
```diff
@@ -14,38 +14,65 @@ export const searchBooks = async (req, res) => {
...
+
+    const booksQuery = await db.query({
+      text: `
+        SELECT b.ISBN, b.Title, b.PublisherName, b.AuthorName, MIN(ub.AskingPrice) AS LowestPrice, MAX(ub.AskingPrice) AS HighestPrice 
+        FROM BOOK AS b
+          JOIN USEDBOOK AS ub ON ub.bookID = b.ISBN
+          LEFT JOIN TEXTBOOK AS tb ON tb.bookID = b.ISBN
+          LEFT JOIN COURSEDEPT AS cd ON cd.serialNumber = tb.serialNumber AND cd.semester = tb.semester
+          LEFT JOIN COURSE AS c ON c.serialNumber = tb.serialNumber AND c.semester = tb.semester
+        WHERE b.Title ILIKE $1
```
Comment (yenting-biao) [inline src/backend/controllers/book.js:42]: ILIKE??

Comment (yenting-biao) [reply src/backend/controllers/book.js:42]: 然後我覺得以正常的搜尋方法來說應該都轉小寫比對

File: src/backend/controllers/book.js [hunk 2]
```diff
@@ -14,38 +14,65 @@ export const searchBooks = async (req, res) => {
...
+      bookQueryValues.push(deptCode);
+      indices["deptCode"] = index++;
+    }
+    if (courseName) {
+      bookQueryValues.push("%" + courseName + "%");
+      indices["courseName"] = index++;
+    }
+    bookQueryValues.push(isbn ? "%" + isbn + "%" : "%");
+    indices["isbn"] = index++;
+
```
Comment (yenting-biao) [inline src/backend/controllers/book.js:33]: Why does index start with 3 rather than 2

Comment (togimoto) [reply src/backend/controllers/book.js:33]: Parameter indices in parameterized queries start from 1. See [https://node-postgres.com/features/queries](https://node-postgres.com/features/queries).

Comment (yenting-biao) [reply src/backend/controllers/book.js:33]: ok

**Review by yenting-biao — APPROVED** [review]
ok to be merged

**Pull request merged**